lib/ClimaCoreSpectra/src/gcm_spectra.jl

```diff
@@ -0,0 +1,392 @@
+# Main implentation based on: https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Common/Spectra/power_spectrum_gcm.jl
+
+#=
+Cleanup items:
+
+ - Inherit array types (i.e., don't assume things are `Array`, as we may have `CuArray` for gpu down the road), instead use `Arr where {Arr <: AbstractArray}`
+ - Follow type promotion rules: https://docs.julialang.org/en/v1/manual/conversion-and-promotion/
+ - Do we need this to be allocation-free?
+ - Can we use external packages (e.g., RootSolvers) for loops that seem to be performing iterations?
+ - Improve documentation
+
+ - Add main module string with todo list:
+  - What API functions are needed?
+    - What are the function argument types?
+    - What do they do / output, in equation form?
```

Comment:
@charleskawczynski some of these remain open questions (for example, at the moment we only employ user-defined parameters and not CC API types/functions). I'd like to connect with you and @LenkaNovak to discuss some of these questions. Thanks

Reply:
I think that these are the questions that need to be answered as a part of this work